### PR TITLE
Docs/shudson fixes

### LINKED ISSUE #232: Add installation section to docs
In addition to whats in the getting started guide, this will provide instructions for different systems - eg. Theta, Summit. It will include spack instructions and should also mention if modules are available on those systems.

Some considerations will be:
Building with Conda, and using intel packages where available.
Installing mpi4py in right way to pick up correct system MPI

Spack install docs will include build instructions for different platforms, links to various yaml files (packages, modules, config) to support and possibly back-end build scripts where relevant.

### PR DESCRIPTION
**Pull request by shuds13**
Documents calc_status and adds Cori guide. I will keep this branch open but I want to bring these changes in.  Should resolve #232

### CODE REVIEW

File: docs/data_structures/calc_status.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
+.. _datastruct-calc-status:
+
+calc_status
+===========
+
+The ``calc_status`` is an integer attribute with a corresponding description that can be used in :ref:`sim_f<api_sim_f>` or :ref:`gen_f<api_gen_f>` functions to capture the status of a calcuation. This is returned to the manager and printed to the ``libE_stats.txt`` file. This is not currently used by the manager, but can still provide a useful summary in libE_stats.txt. The user determines the status of the calculation, as it could include multiple application runs. It can be added as a third return variable in sim_f or gen_f functions.
```
Comment (jmlarson1) [inline docs/data_structures/calc_status.rst]: We say "This is not currently used by the manager" but isn't it used when a worker stops being persistent (on it's own)?

Comment (shuds13) [reply docs/data_structures/calc_status.rst]: Ok, in that case I will also move these into the calc_status section in message_numbers.py and give a description (which will go to libE_stats.txt). I will change this.

Looks like its only used in uniform_or_localopt.py currently.

Comment (shuds13) [reply docs/data_structures/calc_status.rst]: I've changed this now.

File: docs/platforms/cori.rst [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+Debugging Strategies
+--------------------
+
+View the status of your submitted jobs with ``squeue`` and cancel jobs with
+``scancel [Job ID]``.
+
+Additional Information
+----------------------
+
+See the LCRC Bebop docs here_ for more information about Cori.
```
Comment (jlnav) [inline docs/platforms/cori.rst]: "LCRC Bebop" -> "NERSC Cori"

Comment (shuds13) [reply docs/platforms/cori.rst]: Nice spot!

**Pull request merged**